library/Fission/User/Security.hs

```diff
@@ -3,18 +3,24 @@ module Fission.User.Security
   , genID
   ) where
 
-import           Database.Selda (ID)
 import qualified RIO.Text       as Text
 
 import           Fission.Prelude
-import qualified Fission.Random     as Random
+import qualified Fission.Random  as Random
 import           Fission.Security
-import           Fission.User.Types
 
--- | Create a 'SecretDigest' from the users ID
---   Barely an obsfucating technique, but enough to hide DB ordering
-hashID :: ID User -> SecretDigest
-hashID = Text.take 20 . digest
+-- | Generate an ID
+genID :: MonadIO m => m SecretDigest
+genID =
+  50
```

Comment:
such a small gripe, but I feel like the first line of this would read better as `Random.alphaNum 50`

Reply:
Option 1
```haskell
50
  |> Random.alphaNum
  |> fmap hashID
  |> liftIO
```

Option 2
```haskell
liftIO <| hashID <$> Random.alphaNum 50
```

Option 3
```haskell
do
  rando <- Random.alphaNum 50
  liftIO <| return <| hashID rando
```

Option 4
```haskell
do
  rando <- Random.alphaNum 50
  rando
    |> hashID
    |> return
    |> liftIO
```

Option 5
```haskell
liftIO . return . hashID =<< Random.alphaNum 50
```

Thoughts, @dholms ?

Reply:
ok yeah I like option 1 & 2 the best out of those.

And the reason for not doing this?
```
  Random.alphaNum 50
    |> fmap hashID
    |> liftIO
```

Is it just against convention to put the first function before the pipes?
My brain interprets pipes as "take this value and apply these transformations to it" and `50` seems so arbitrary that at first glance it doesn't make much sense. `Random.alphaNum 50` gives way more context to what the value being transformed actually is

[ReasonML](https://reasonml.github.io/docs/en/pipe-first) doesn't turn the innermost function into a pipe :stuck_out_tongue_winking_eye:

Reply:
I personally like Options 2, but that's a decade for idiomatic Haskell rearing its head 😅

> Is it just against convention to put the first function before the pipes?
Yeah, this is OCaml/Reason vs Elixir convention 😜

The idea is that since we're relying on a directional metaphor, we should stick to one direction. This is all convention, and Haskell doesn't normally have pipes... so we need to decide what we want to set as a style here.

I used to teach Elixir, so "go with the flow" (direction) is pretty drilled into me, but whatever we can change. I can see an argument especially with nested constructors.

Thoughts @fission-suite/development?

Reply:
Oops, @icidasset wasn't on that list, so pinging separately